### DISCUSSION TITLE
If Possible could you roll the changes of a fork into the main build?

### DISCUSSION THREAD

**Discussion by Vincevinny** [lllyasviel/Fooocus#1726] [Ideas]
hey again mate hows things?
love the face-swap added BTW genuinely love this interface
but wanted a bit more control on in-painting/image manipulation and this guys fork https://github.com/fenneishi/Fooocus-ControlNet-SDXL
add adds a fair bit of the new CN-XL stuff but as its not the main branch its going to always lag behind your version and i would prefer to not have 3 forks installed at once if possible,
i have genuinely no idea how much work this is or if its against TOS of Git or something so ignore me if im dumb

**Comment by eddyizm** [comment]
Why don't you just use that fork and pull the upstream changes in from the original repo (this one) ? that's what I do with mine. 
Just went to go check it out and it seems to be , 213 commits behind, so I would do this carefully in a branch as you may have to deal with merge conflicts.

**Reply by Vincevinny** [reply]
did not even know about this feature and will look into it thanks heaps!

**Comment by mashb1t** [comment]
Please find the forking guidelines here: https://github.com/lllyasviel/Fooocus/discussions/699#discussion-5740850

I know the core team(members) of Fooocus have gone a bit dark, but they have announced to be back at mid/end January.
This could also mean that features are then migrated back from downstream to Fooocus again, potentially including more Controlnet support.

**Reply by Vincevinny** [reply]
legend thanks!
